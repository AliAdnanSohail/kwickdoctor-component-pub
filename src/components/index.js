export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
