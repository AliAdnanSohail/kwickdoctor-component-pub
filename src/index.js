export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
