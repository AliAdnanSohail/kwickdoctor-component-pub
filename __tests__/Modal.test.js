import React, { Fragment } from 'react';
import { mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';

import { Modal, Button } from '../src';

const baseProps = {
  active: true,
  rootId: 'modalRoot',
  onExit: () => {},
};

const modalProps = {
  className: 'test-modal-content',
  text: 'test modal content',
};

class ModalWrapper extends React.Component {
  constructor(props) {
    super(props);

    this.state = { active: props.active };
  }

  toggleModal = (state) => {
    this.setState({ active: state });
  };

  render() {
    const { escapeExits, underlayClickExits } = this.props;
    const { active } = this.state;

    return (
      <Fragment>
        <Button
          onClick={() => {
            this.toggleModal(true);
          }}
        >
          Open modal
        </Button>
        <Modal
          escapeExits={escapeExits}
          underlayClickExits={underlayClickExits}
          active={active}
          onExit={() => {
            this.toggleModal(false);
          }}
        >
          <ModalContent {...modalProps} />
        </Modal>
      </Fragment>
    );
  }
}

const ModalContent = props => <div className={props.className}>{props.text}</div>;

const TestModal = () => (
  <div id={baseProps.rootId}>
    <Modal {...baseProps}>
      <ModalContent {...modalProps} />
    </Modal>
  </div>
);

it('renders correctly', () => {
  const element = mount(<TestModal />);

  expect(EnzymeToJson(element)).toMatchSnapshot();
});

it('modal content renders correctly', () => {
  const element = mount(<ModalWrapper active />);

  expect(element.find('.modal__content').text()).toEqual(modalProps.text);
});

it('shows correctly', () => {
  const element = mount(<ModalWrapper active={false} />);
  element.find('button').simulate('click');

  expect(element.find('.modal__content')).toHaveLength(1);
});

it('hides correctly', () => {
  const element = mount(<ModalWrapper active />);
  element.find('.modal').simulate('click');

  setTimeout(() => {
    expect(element.find('.modal').exists()).toEqual(false);
  }, 300);
});

it('underlay click event works correctly', () => {
  const element = mount(<ModalWrapper active />);
  element.find('.underlay').simulate('mousedown');

  setTimeout(() => {
    expect(element.find('.modal').exists()).toEqual(false);
  }, 300);
});

it('disabled underlay click event works correctly', () => {
  const element = mount(<ModalWrapper active escapeExits={false} />);
  element.find('.underlay').simulate('mousedown');

  setTimeout(() => {
    expect(element.find('.modal').exists()).toEqual(true);
  }, 300);
});

it('esc button works correctly', () => {
  const element = mount(<ModalWrapper active />);
  element.simulate('keyDown', { keyCode: 27 });

  setTimeout(() => {
    expect(element.find('.modal').exists()).toEqual(false);
  }, 300);
});

it('disabled esc button works correctly', () => {
  const element = mount(<ModalWrapper active escapeExits={false} />);
  element.simulate('keyDown', { keyCode: 27 });

  setTimeout(() => {
    expect(element.find('.modal').exists()).toEqual(true);
  }, 300);
});
