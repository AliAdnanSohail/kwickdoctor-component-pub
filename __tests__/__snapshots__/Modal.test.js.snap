// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ModalWrapper
  active={false}
>
  <Button
    accent={false}
    className=""
    danger={false}
    disabled={false}
    flat={false}
    icon={null}
    loading={false}
    onClick={[Function]}
    rounded={false}
    squared={false}
    transparent={false}
    type="button"
  >
    <button
      className="jsx-1748154264 button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="button__content"
      >
        Open modal
      </span>
      <JSXStyle
        css={
          Array [
            ".button.jsx-1748154264{position:relative;padding:13px 64px;height:52px;background-color:#0c97f9;box-sizing:border-box;cursor:pointer;color:#ffffff;line-height:26px;-webkit-transition:all 0.4s ease-in-out;transition:all 0.4s ease-in-out;overflow:hidden;border-width:1px;border-style:solid;border-color:#0c97f9;border-radius:26px;font-style:normal;font-weight:500;font-size:1rem;text-align:center;text-transform:uppercase;}",
            ".button.jsx-1748154264:focus{outline:none;}",
            ".button.jsx-1748154264 svg{fill:#ffffff;stroke:#ffffff;width:24px;height:24px;}",
            ".button.jsx-1748154264 svg circle,path{stroke:#ffffff;}",
            ".button__content{display:inline-block;-webkit-transition:all 0.3s ease-in-out;transition:all 0.3s ease-in-out;}",
            ".button__wave.jsx-1748154264{position:absolute;display:block;width:0;height:0;border-radius:50%;background-color:#ffffff;opacity:0.5;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
            ".button--loading.jsx-1748154264 .button__content{-webkit-transform:translateX(16px);-ms-transform:translateX(16px);transform:translateX(16px);}",
            ".button--disabled.jsx-1748154264{opacity:0.7;cursor:not-allowed;}",
            ".button.jsx-1748154264:not(.button--disabled):hover,.button.jsx-1748154264:not(.button--disabled):focus{background-color:#38a9fa;border-color:#38a9fa;box-shadow:2px 2px 10px rgba(0,0,0,0.1);}",
            ".button--flat.jsx-1748154264{color:#0c97f9;background-color:transparent;border-color:transparent;}",
            ".button--flat.jsx-1748154264 svg{stroke:#2A2A30;fill:#2A2A30;}",
            ".button--flat.jsx-1748154264 svg path,circle{stroke:#2A2A30;}",
            ".button--flat.jsx-1748154264:not(.button--disabled):hover,.button--flat.jsx-1748154264:not(.button--disabled):focus{color:#0c97f9;background-color:transparent;border-color:transparent;box-shadow:none;}",
            ".button--accent.jsx-1748154264{color:#0c97f9;background-color:transparent;border-color:transparent;}",
            ".button--accent.jsx-1748154264 svg{stroke:#2A2A30;fill:#2A2A30;}",
            ".button--accent.jsx-1748154264 svg path,circle{stroke:#2A2A30;}",
            ".button--accent.jsx-1748154264:not(.button--disabled):hover,.button--accent.jsx-1748154264:not(.button--disabled):focus{color:#0c97f9;background-color:transparent;border-color:transparent;}",
            ".button--danger.jsx-1748154264{color:#ffffff;background-color:#ea6577;border-color:#ea6577;}",
            ".button--danger.jsx-1748154264:not(.button--disabled):hover,.button--danger.jsx-1748154264:not(.button--disabled):focus{background-color:#ed7888;border-color:#ed7888;}",
            ".button--rounded.jsx-1748154264{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;width:52px;height:52px;padding:0 !important;border-radius:50%;}",
            ".button.jsx-1748154264:not(.button--rounded) .button__loading-icon{position:absolute;top:calc(50% - 12px);left:32px;}",
            ".button--squared.jsx-1748154264{border-radius:12px;}",
            ".button--transparent.jsx-1748154264{background-color:transparent;box-shadow:0px 0px 10px rgba(12,151,249,0.15);color:#2A2A30;}",
            ".button--transparent.jsx-1748154264:not(.button--disabled):hover,.button--transparent.jsx-1748154264:not(.button--disabled):focus{color:#ffffff;}",
            ".button--transparent.jsx-1748154264 svg{stroke:#2A2A30;fill:#2A2A30;}",
            ".button--transparent.jsx-1748154264 svg path,circle{stroke:#2A2A30;}",
            ".button--xs.jsx-1748154264{height:32px;line-height:1;padding:0 32px;font-size:0.75rem;}",
            ".button--xs.button--rounded.jsx-1748154264{width:32px;height:32px;}",
            ".button--xs.jsx-1748154264 svg{width:18px;height:18px;}",
            ".button--xs.jsx-1748154264:not(.button--xs--rounded) .button__loading-icon{left:16px;top:calc(50% - 9px);}",
            ".button--s.jsx-1748154264{height:42px;line-height:1;padding:0 42px;font-size:0.875rem;}",
            ".button--s.button--rounded.jsx-1748154264{width:42px;height:42px;}",
            ".button--s.jsx-1748154264 svg{width:18px;height:18px;}",
            ".button--s.jsx-1748154264:not(.button--s--rounded) .button__loading-icon{left:21px;top:calc(50% - 9px);}",
            ".button.has-clicked.jsx-1748154264 .button__wave.jsx-1748154264{-webkit-animation:wave-jsx-1748154264 0.8s;animation:wave-jsx-1748154264 0.8s;}",
            "@-webkit-keyframes wave-jsx-1748154264{0%{opacity:0.5;width:0;height:0;}100%{opacity:0;width:300px;height:300px;}}",
            "@keyframes wave-jsx-1748154264{0%{opacity:0.5;width:0;height:0;}100%{opacity:0;width:300px;height:300px;}}",
          ]
        }
        styleId="1748154264"
      />
    </button>
  </Button>
  <Modal
    active={false}
    alert={false}
    escapeExits={true}
    onExit={[Function]}
    rootId="root"
    title="dom one"
    underlayClickExits={true}
  >
    <Displaced
      alert={false}
      escapeExits={true}
      getApplicationNode={[Function]}
      mounted={false}
      onEnter={[Function]}
      onExit={[Function]}
      titleText="dom one"
      underlayClass="underlay"
      underlayClickExits={true}
      underlayColor={false}
      verticallyCenter={true}
    />
  </Modal>
</ModalWrapper>
`;
